Tools/autotest/arducopter.py

```diff
@@ -1671,33 +1671,121 @@ def fly_optical_flow_limits(self):
+            self.wait_mode('AUTOTUNE')
+
+            tstart = self.get_sim_time()
+            sim_time_expected = 5000
+            deadline = tstart + sim_time_expected
+            while self.get_sim_time_cached() < deadline:
+                now = self.get_sim_time_cached()
+                m = self.mav.recv_match(type='STATUSTEXT',
+                                        blocking=True,
+                                        timeout=1)
+                if m is None:
+                    continue
+                self.progress("STATUSTEXT (%u<%u): %s" % (now, deadline, m.text))
+                if "AutoTune: Success" in m.text:
+                    self.progress("AUTOTUNE OK (%u seconds)" % (now - tstart))
+                    # Check original gains are re-instated
+                    self.set_rc(8, 1100)
+                    self.delay_sim_time(1)
+                    self.progress("AUTOTUNE original gains are P:%f I:%f D:%f" % (self.get_parameter("ATC_RAT_RLL_P"), self.get_parameter("ATC_RAT_RLL_I"), self.get_parameter("ATC_RAT_RLL_D")))
+                    if rlld != self.get_parameter("ATC_RAT_RLL_D") or rlli != self.get_parameter("ATC_RAT_RLL_I") or rllp != self.get_parameter("ATC_RAT_RLL_P"):
+                        raise NotAchievedException("AUTOTUNE gains still present")
+                    # Use autotuned gains
+                    self.set_rc(8, 1850)
+                    self.delay_sim_time(1)
+                    self.progress("AUTOTUNE testing gains are P:%f I:%f D:%f" % (self.get_parameter("ATC_RAT_RLL_P"), self.get_parameter("ATC_RAT_RLL_I"), self.get_parameter("ATC_RAT_RLL_D")))
+                    if rlld == self.get_parameter("ATC_RAT_RLL_D") or rlli == self.get_parameter("ATC_RAT_RLL_I") or rllp == self.get_parameter("ATC_RAT_RLL_P"):
+                        raise NotAchievedException("AUTOTUNE gains not present in pilot testing")
+                    # land without changing mode
+                    self.set_rc(3, 1000)
+                    self.wait_for_alt(0)
+                    self.wait_disarmed()
+                    # Check gains are still there after disarm
+                    if rlld == self.get_parameter("ATC_RAT_RLL_D") or rlli == self.get_parameter("ATC_RAT_RLL_I") or rllp == self.get_parameter("ATC_RAT_RLL_P"):
+                        raise NotAchievedException("AUTOTUNE gains not present on disarm")
+
+                    self.reboot_sitl()
+                    # Check gains are still there after reboot
+                    if rlld == self.get_parameter("ATC_RAT_RLL_D") or rlli == self.get_parameter("ATC_RAT_RLL_I") or rllp == self.get_parameter("ATC_RAT_RLL_P"):
+                        raise NotAchievedException("AUTOTUNE gains not present on reboot")
+                    # Check FLTT is unchanged
+                    if rllt != self.get_parameter("ATC_RAT_RLL_FLTT"):
+                        raise NotAchievedException("AUTOTUNE FLTT was modified")
+                    return
```

Comment:
Again, suspicions use of `return` :-)

Reply:
The control flow is correct and what I inherited from the previous test. Removing return means rewriting the test which seems over the top